trin-bridge/src/types.rs

```diff
@@ -0,0 +1,32 @@
+use std::fmt;
+use std::str::FromStr;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum NetworkKind {
```

Comment:
A bit of a nitpick, but I'd prefer something like `Subnetwork` here - not a blocker, so feel free to ignore me. I just hate types that contain `Type` / `Kind` in the name.